docs/rules.txt

```diff
@@ -0,0 +1,93 @@
+rules_version = '2';
+service cloud.firestore {
+  match /databases/{database}/documents {
+     function signedIn() {
+      return request.auth.uid != null;
+    }
+
+    function isDocOwner() {
+      return request.auth.uid == resource.data.owner;
+    }
+
+    function updatedDocHasCorrectOwner() {
+      return request.auth.uid == request.resource.data.owner;
+    }
+
+    function updatedDocHasSameOwner() {
+      return resource.data.owner == request.resource.data.owner;
+    }
+
+    function updatedDocHasSameSharing() {
+      return resource.data.sharedWith == request.resource.data.sharedWith;
+    }
+
+    function isSharedWithMe() {
+      return request.auth.token.email in resource.data.sharedWith;
+    }
+
+    function isSharedOnlyWithMe() {
+    	return [request.auth.token.email] == request.resource.data.sharedWith;
+    }
+
+    function isVerified() {
+    	return request.auth.token.email_verified;
+    }
+
+    function stillSharedWithMe() {
+    	return request.auth.token.email in request.resource.data.sharedWith;
+    }
+
+    match /Lists-AuthenticationRequired/{list} {
+      allow read: if signedIn() && isDocOwner();
+      allow create: if signedIn()&& updatedDocHasCorrectOwner();
+      allow update: if signedIn() && isDocOwner() &&
+        updatedDocHasCorrectOwner();
+      allow delete: if signedIn() && isDocOwner();
+      	match /Tasks-AuthenticationRequired/{task} {
+      		allow read: if signedIn() && isDocOwner();
+      		allow create: if signedIn() && updatedDocHasCorrectOwner();
+      		allow update: if signedIn() && isDocOwner() &&
+        		updatedDocHasCorrectOwner();
+      		allow delete: if signedIn() && isDocOwner();
+    	}
+    }
+
+    isSharedWithMe_Task(list){
+        get(/database/$(database)/documents/lists/$(list).token.email in resource.data.sharedWith
+    }
+
+    isVerified_Task(list) {
+        get(/database/$(database)/documents/lists/$(list).token.email_verified
+    }
+
+    isDocOwner_Task(list) {
+        request.auth.uid == get(/database/$(database)/documents/lists/$(list).data.owner;
+    }
+
+    updatedDocHasCorrectOwner_Task(list) {
+        return request.auth.uid == get(/database/$(database)/documents/lists/$(list).data.owner;
+    }
+
+    updatedDocHasSameOwner_Task(list) {
+        return get(/database/$(database)/documents/lists/$(list).data.owner == request.get(/database/$(database)/documents/lists/$(list).data.owner;
```

Comment:
No need for this. If the user is updating a task document, there's no way they are modifying a list document.